appscale/tools/remote_helper.py

```diff
@@ -940,41 +942,46 @@ def terminate_virtualized_cluster(cls, keyname, is_verbose):
 
     acc = AppControllerClient(shadow_host, secret)
     try:
-      all_ips = acc.get_all_public_ips()
+      terminate_thread = threading.Thread(target=acc.run_terminate,
```

Comment:
In the interest in making things simpler for the client, I would recommend making this asynchronous on the server side since you are already polling on the client side. In other words, the server function should return immediately (doing the work in a separate thread), and the client should not need to use threads.